app/src/lib/api.ts

```diff
@@ -2147,6 +2159,15 @@ export function getHTMLURL(endpoint: string): string {
  * http://github.mycompany.com -> http://github.mycompany.com/api/v3
  */
 export function getEnterpriseAPIURL(endpoint: string): string {
+  if (isGHE(endpoint)) {
```

Comment:
Is the (legitimate) GHES case covered here? The path prefix is still used in GHES, but that's the only place where it's allowed. 

Basically anything in dotcom and anything with a `.ghe.com` domain should use the `api.` subdomain.

Reply:
I think that's correct: https://github.com/desktop/desktop/blob/45f5c1e37de62da0c707b2ce246c92b34069ab5c/app/src/lib/endpoint-capabilities.ts/#L61-L63

```typescript
/** Whether or not the given endpoint URI is under the ghe.com domain */
export const isGHE = (ep: string) => new URL(ep).hostname.endsWith('.ghe.com')
```

I tested it and works as you describe. dotcom and `.ghe.com` requests use the `api.` subdomain while the rest still use `/api/v3`.

The only thing I've noticed is that existing logged-in users of `.ghe.com` domains will still use `/api/v3` because we persist that base URL for users, and therefore those would need to be migrated:

```json
[
  {
    "login": "sergiou87",
    "endpoint": "https://api.github.com",
    ...
  },
  {
    "login": "sergiou87",
    "endpoint": "https://whatever.ghe.com/api/v3",
    ...
  }
]
```

We will take care of that in a different PR, but I will keep you posted!

Reply:
Got it, thanks for the confirmation!